Background layer crash

Trying to draw in the background layer crashes hard. 

Exception in thread "LWJGL Application" com.badlogic.gdx.utils.GdxRuntimeException: java.lang.OutOfMemoryError: Java heap space
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:127)
Caused by: java.lang.OutOfMemoryError: Java heap space

--- a/jump-core/src/main/java/com/bitdecay/jump/level/SingleLayer.java
+++ b/jump-core/src/main/java/com/bitdecay/jump/level/SingleLayer.java
@@ -17,8 +17,12 @@ public class SingleLayer {
     }
 
     public SingleLayer(int cellSize) {
+        this(cellSize, 10, 10);
+    }
+
+    public SingleLayer(int cellSize, int width, int height) {
         this.cellSize = cellSize;
-        grid = new TileObject[10][10];
+        grid = new TileObject[width][height];
         otherObjects = new HashMap<>();
         triggers = new HashMap<>();
     }

--- a/jump-core/src/main/java/com/bitdecay/jump/level/builder/LevelLayersBuilder.java
+++ b/jump-core/src/main/java/com/bitdecay/jump/level/builder/LevelLayersBuilder.java
@@ -73,7 +73,7 @@ public class LevelLayersBuilder implements ILevelBuilder {
         activeLayer = layer;
 
         if (!activeLevel.layers.hasLayer(layer)) {
-            SingleLayer newLayer = new SingleLayer(activeLevel.layers.cellSize);
+            SingleLayer newLayer = new SingleLayer(activeLevel.layers.cellSize, activeLevel.layers.getLayer(0).grid.length, activeLevel.layers.getLayer(0).grid[0].length);
             activeLevel.layers.addLayer(layer, newLayer);
         }
 

--- a/jump-leveleditor/src/main/java/com/bitdecay/jump/leveleditor/render/LevelEditor.java
+++ b/jump-leveleditor/src/main/java/com/bitdecay/jump/leveleditor/render/LevelEditor.java
@@ -250,12 +250,12 @@ public class LevelEditor extends InputAdapter implements Screen, OptionsUICallba
         }
         shaper.setColor(BitColors.GRID_SIZE);
 
-        TileObject[][] grid = curLevelBuilder.getLevel().layers.getLayer(0).grid;
-        shaper.rect(curLevelBuilder.getLevel().layers.gridOffset.x * curLevelBuilder.getCellSize(), curLevelBuilder.getLevel().layers.gridOffset.y * curLevelBuilder.getCellSize(),
-                grid.length * curLevelBuilder.getCellSize(), grid[0].length * curLevelBuilder.getCellSize());
+        for (Integer layerNumber : curLevelBuilder.getLevel().layers.layers.keySet()) {
+            TileObject[][] grid = curLevelBuilder.getLevel().layers.getLayer(layerNumber).grid;
+            shaper.rect(curLevelBuilder.getLevel().layers.gridOffset.x * curLevelBuilder.getCellSize(), curLevelBuilder.getLevel().layers.gridOffset.y * curLevelBuilder.getCellSize(),
+                    grid.length * curLevelBuilder.getCellSize(), grid[0].length * curLevelBuilder.getCellSize());
+        }
         shaper.end();
-
-
     }
 
     private void renderStrings() {

